### PR TITLE
fix(searchBox): pass "spellcheck" property correctly to input

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

React and Preact differ slightly in how they pass attributes / properties. Therefore it seems like the correct value (although https://github.com/preactjs/preact/issues/485 says otherwise, that must have changed in preact X). This was causing `spellcheck` not to be applied.

see: https://github.com/preactjs/preact/blob/020d5e599b8b84745932f092a835f494b1dd2261/src/jsx.d.ts#L711

ref: https://discourse.algolia.com/t/how-to-add-spellcheck-property-to-search-box-widget/10741

**Result**

The `spellcheck` property is actually on the dom node for input.

I have checked, and the other properties on the input are present.

### CODE REVIEW

File: src/components/SearchBox/SearchBox.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ class SearchBox extends Component {
             autoComplete="off"
             autoCorrect="off"
             autoCapitalize="off"
-            spellCheck={false}
+            spellcheck={false}
```
Comment (francoischalifour) [inline src/components/SearchBox/SearchBox.js:162]: I had the same bug in the Preact export of DocSearch and the actual fix is to use:

```suggestion
            spellCheck="false"
```

Comment (Haroenv) [reply src/components/SearchBox/SearchBox.js:162]: interesting, my fix seems to work too, but I'll try your suggestion

File: src/components/SearchBox/__tests__/SearchBox-test.js [hunk 2]
```diff
@@ -438,6 +438,20 @@ describe('SearchBox', () => {
       expect(mount(<SearchBox {...defaultProps} />)).toMatchSnapshot();
     });
 
+    test('sets autocorrect... properties', () => {
```
Comment (francoischalifour) [inline src/components/SearchBox/__tests__/SearchBox-test.js:441]:
```suggestion
    test('sets search input properties', () => {
```

File: src/components/SearchBox/__tests__/SearchBox-test.js [hunk 3]
```diff
@@ -438,6 +438,20 @@ describe('SearchBox', () => {
...
+      const res = render(
+        <SearchBox {...defaultProps} autofocus={true} query="sample query" />
+      );
+      const input = res.getByDisplayValue('sample query');
+
+      expect(input.getAttribute('autofocus')).toBe('true');
```
Comment (francoischalifour) [inline src/components/SearchBox/__tests__/SearchBox-test.js:447]: We have helpers for DOM elements in this file. Did you try them?

```suggestion
      expect(input).toHaveAttribute('autofocus', 'true');
```

File: .eslintrc.js [hunk 4]
```diff
@@ -2,24 +2,24 @@ module.exports = {
   extends: ['algolia', 'algolia/jest', 'algolia/react', 'algolia/typescript'],
   plugins: ['react-hooks'],
   rules: {
-    'no-param-reassign': 0,
-    'import/no-extraneous-dependencies': 0,
+    'no-param-reassign': 'off',
```
Comment (francoischalifour) [inline .eslintrc.js:5]: Any particular reasons this file was changed?

Comment (Haroenv) [reply .eslintrc.js:5]: I originally changed it for adding the `spellcheck` exception, and made the file consistent about 0/off then

File: src/components/SearchBox/__tests__/SearchBox-test.js (hunk 3 above)
Comment (Haroenv) [reply src/components/SearchBox/__tests__/SearchBox-test.js:447]: did not know about those, thanks

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**